packages/geo/src/lib/layer/layer-legend/layer-legend.component.ts

```diff
@@ -27,5 +28,25 @@ export class LayerLegendComponent {
   }
   private _legend;
 
-  constructor() {}
+  constructor(private capabilitiesService: CapabilitiesService) {}
+
+   computeItemTitle(item): string {
+    const layerOptions = this.layer.dataSource.options as any;
+    if (layerOptions.type === 'wms' && layerOptions.optionsFromCapabilities) {
+      let localLayerOptions = JSON.parse(JSON.stringify(layerOptions)); // to avoid to alter the original options.
+      layerOptions.params.layers.split(',').forEach(layer => {
+        if (layer === item.title) {
+          localLayerOptions.params.layers = layer;
+          this.capabilitiesService.getWMSOptions(localLayerOptions).subscribe(r => localLayerOptions = r);
```

Comment:
This runs asynchronously but your method doesn't return an observable.

Reply:
This still doesn't return an observable. You might not want to return an observable but, in that case, you can't subscribe to getWMSOptions (which makes a GetCapabilities) and assume that it'll complete in time for you to return localLayerOptions._layerOptionsFromCapabilities.title directly.